### ISSUE TITLE
Autcomplete of only java-ide not working properly

### ISSUE THREAD

**Issue by KJ1010G** [atom/ide-java#84]
*Edit by @rsese to add repro steps based on https://github.com/atom/ide-java/issues/75#issuecomment-396820888*

1. Fresh install latest version of Atom 
2. Install the latest version of atom-ide-ui.
3. Install the latest version of ide-java.
4. Create and save a file called `Test.java` with this content:

```java
import javax.swing.*;

public class Test {
	public static void main(String[] args) {
		JFrame frame = new JFrame();
		frame.
	}
}
```
5. After the `.` in `frame.`, type `setM` to bring up the autocomplete menu.
6. Select the first item using <kbd>enter</kbd> or by mouse clicking.

Expect the item selected to autocomplete, actual is the autocomplete menu is dismissed.  Not 100% of the time but hard to put a number on it.

---

Note - This only started happening after installing atom-ide-ui & ide-java

### Description
When i am typing some code, for example-
frame.setVi
at this point it would correctly show me the auto-complete options.
![issue_java_ide_atom](https://user-images.githubusercontent.com/38991076/41072835-d1553ffa-6a1d-11e8-9cbe-b89a17b835c9.png)
No problems till now.

### Expected Behavior
Now, when I press TAB/ENTER to auto-complete the highlighted option ( setVisible(Boolean b) in this example ) , it should correctly auto-complete in my code.

### Actual Behavior
But it doesn't approximately 70% of time. When i press TAB/ENTER, the options diss-appear.
![issue_java_ide_atom_2](https://user-images.githubusercontent.com/38991076/41073092-384ef27c-6a1f-11e8-9cbe-e11843d381c0.png)
Then i have to press, Ctrl+Space to make the options visible again. And choose the same option( if I choose some other and press TAB, then too nothing happens) again and then press TAB. Then it auto-completes.
Basically, 30% of time, auto-complete works when I press TAB on an option once.
And, 100% of time, auto-complete works when I press TAB on an option twice.
So, 70% (approx) of time i have to press TAB twice.

Also, this never happens when the option is of "language-java" package. The default atom package.
Those options get Auto-completed in first attempt always.

### Versions
* Atom: 1.27.2
* Client OS: Ubuntu 16.04LTS

### Additional Details
* Installed packages:
- atom-ide-ui 0.12.0
- ide-java 0.8.3 
- and core packages, nothing else

**Comment by adrianmalacoda** [comment]
I've experienced this issue too. It's apparently due to a conflict with `language-java` package also doing autocomplete. See #74, #75

**Comment by sadikovi** [comment]
This issue does not seem to be related in any way to the `language-java`. It is about what keys are set for autocomplete (_pretty reasonable assumption, tested with using dev version of package and removing any related to completion settings_).

I tested on OS X, atom v 1.27.2 (normal and dev modes). Here is the list of observations and potential fixes that I found working.

- By default, assuming that a person did not change `autocomplete-plus` settings, ide-java autocomplete always works when using TAB key, regardless of selecting the first item in the list or using arrows to select the other. 
- Autocomplete does not work with ENTER key.
- Autocomplete works properly if you click on the item that is highlighted in the list - the entire expression is inserted into the code.
- Autocomplete does not work properly if you click on the item that is not currently highlighted - it only inserts the function name without parameters.

To fix the selection with TAB/ENTER, go to `autocomplete-plus` settings and change the value of "Keymap for Confirming a Suggestion" option, e.g. `tab and enter`, then autocomplete selection works as expected (though it may affect other autocomplete selections).

To solve "select with click" issue... Well, it works if you click on selected item in the list, e.g. moving with arrows and then clicking on a selected item. However, it only inserts the function name, if you click on non-highlighted item. I am not sure how critical this is, but it looks like it could be fixed in this package by opening a PR.

**Comment by sadikovi** [comment]
By the way, it is exactly the same behaviour in language-java package to suggest function names, variables, etc.

**Comment by sadikovi** [comment]
As we discussed, we might need to debug it further - I will continue debugging with different packages, see if it is actually consistent to reproduce.

**Comment by sadikovi** [comment]
Could be related to https://github.com/atom/autocomplete-plus/issues/975

**Comment by rsese** [comment]
This is fixed by https://github.com/atom/autocomplete-plus/issues/975 and the fix will be available in an upcoming release, thanks for investigating @sadikovi!
